apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -420,14 +420,14 @@ def write(
 
                     enmr = self._handle.enum(attr.name)
 
-                    # get new enumeration values, maintain original ordering
-                    update_vals = []
-                    for new_val in col.chunk(0).dictionary.tolist():
-                        if new_val not in enmr.values():
-                            update_vals.append(new_val)
+                    # get new enumeration values by taking the set difference
+                    # while maintaining ordering
+                    update_vals = np.setdiff1d(
+                        col.dictionary, enmr.values(), assume_unique=True
+                    )
 
                     # only extend if there are new values
-                    if update_vals:
+                    if len(update_vals) != 0:
```

Comment:
nit: this can be `if update_vals: ...` as arrays are truthy when non-zero length